v2/yarpcgrpc/dialer.go

```diff
@@ -22,32 +22,124 @@ package grpc
 
 import (
 	"context"
+	"fmt"
+	"math"
 	"sync"
 
+	"github.com/opentracing/opentracing-go"
 	"go.uber.org/multierr"
+	backoffapi "go.uber.org/yarpc/api/backoff"
+	"go.uber.org/yarpc/internal/backoff"
 	"go.uber.org/yarpc/v2"
 	"go.uber.org/yarpc/v2/yarpcpeer"
+	"go.uber.org/yarpc/yarpcconfig"
+	"go.uber.org/zap"
+	"google.golang.org/grpc"
+	"google.golang.org/grpc/credentials"
+)
+
+const (
+	// defensive programming: these are copied from grpc-go but we set them
+	// explicitly here in case these change in grpc-go so that YARPC stays
+	// consistent.
+	defaultClientMaxRecvMsgSize = 1024 * 1024 * 4
+	defaultClientMaxSendMsgSize = math.MaxInt32
 )
 
 var _ yarpc.Dialer = (*Dialer)(nil)
 
-// Dialer is a decorator for a gRPC transport that threads dial options for
-// every retained peer.
+// Dialer keeps track of all gRPC peers
 type Dialer struct {
+	// ClientMaxRecvMsgSize is the maximum message size the client can receive.
+	//
+	// The default is 4MB.
+	ClientMaxRecvMsgSize int
+	// ClientMaxSendMsgSize is the maximum message size the client can send.
+	//
+	// The default is unlimited.
+	ClientMaxSendMsgSize int
+
+	// BackoffStrategy specifies the backoff strategy for delays between
+	// connection attempts for each peer.
+	//
+	// The default is exponential backoff starting with 10ms fully jittered,
+	// doubling each attempt, with a maximum interval of 30s. //TODO(apeatsbond)
+	Backoff yarpcconfig.Backoff
+
+	// Credentials specifies connection level security credentials (e.g.,
+	// TLS/SSL) for outbound connections.
+	Credentials credentials.TransportCredentials
+
+	// Tracer configures a logger for the dialer.
+	Logger *zap.Logger
+
+	// Tracer configures a tracer for the dialer.
+	Tracer opentracing.Tracer
+
+	internal *dialerInternals
+}
+
+type dialerInternals struct {
```

Comment:
Is the `dialerInternals` struct primarily used to separate the un-exported fields from the exported fields? If so, should we consider simply flattening the `dialerInternals` into the `Dialer`? 

If there is another reason for the separation here, let's consider adding a note describing the significance of this type.

Reply:
noted later, also reflects the `yarpchttp` package for consistency